File: attendance_register/crop_icon.dart
import 'dart:io';
import 'package:image/image.dart' as img;

void main() {
  final file = File('assets/icon/icon.png');
  final image = img.decodeImage(file.readAsBytesSync())!;
  
  int left = image.width;
  int top = image.height;
  int right = 0;
  int bottom = 0;
  
  for (int y = 0; y < image.height; y++) {
    for (int x = 0; x < image.width; x++) {
      final pixel = image.getPixel(x, y);
      if (pixel.r < 245 || pixel.g < 245 || pixel.b < 245) {
        if (x < left) left = x;
        if (y < top) top = y;
        if (x > right) right = x;
        if (y > bottom) bottom = y;
      }
    }
  }
  
  if (right >= left && bottom >= top) {
    // Rimuoviamo completamente il padding per farlo enorme
    int padding = 0; 
    left = (left - padding).clamp(0, image.width);
    top = (top - padding).clamp(0, image.height);
    right = (right + padding).clamp(0, image.width);
    bottom = (bottom + padding).clamp(0, image.height);
    
    final cropped = img.copyCrop(image, x: left, y: top, width: right - left, height: bottom - top);
    
    // Su Android Adaptive gli sfondi trasparenti o bianchi vengono gestiti dalla mask.
    // Salviamolo come png e impostiamolo per l'app.
    File('assets/icon/icon_cropped.png').writeAsBytesSync(img.encodePng(cropped));
    print('Cropped successfully!');
  } else {
    print('Could not crop.');
  }
}

File: attendance_register/crop_icon.py
from PIL import Image

def crop_white_borders(image_path, output_path):
    img = Image.open(image_path).convert("RGBA")
    data = img.load()
    width, height = img.size
    
    # Trova i confini non bianchi
    left, top, right, bottom = width, height, 0, 0
    for y in range(height):
        for x in range(width):
            r, g, b, a = data[x, y]
            # Consider white background
            if not (r > 240 and g > 240 and b > 240):
                if x < left: left = x
                if y < top: top = y
                if x > right: right = x
                if y > bottom: bottom = y

    # Aggiungi un piccolo margine per non tagliare troppo a vivo, o no, tagliamo a vivo!
    if right >= left and bottom >= top:
        # Per Android Adaptive Icon, il foreground dovrebbe riempire bene lo spazio.
        cropped = img.crop((left, top, right, bottom))
        # Ridimensiona a 1024x1024 se si vuole, o lascia così
        cropped.save(output_path)
        print("Cropped successfully from", (left, top, right, bottom))
    else:
        print("Could not crop.")

crop_white_borders("assets/icon/icon.png", "assets/icon/icon_cropped.png")
